CI failure: CMake 4.0 or higher is required

### Problem description

```
CMake Error at CMake/resolve_dependency_modules/cudf.cmake:18 (cmake_minimum_required):
  CMake 4.0 or higher is required.  You are running version 3.31.1
Call Stack (most recent call first):
  CMake/ResolveDependency.cmake:42 (include)
  CMake/ResolveDependency.cmake:83 (velox_build_dependency)
  CMakeLists.txt:533 (velox_resolve_dependency)
```

### System information

NaN

### CMake log

```bash
https://github.com/facebookincubator/velox/actions/runs/29914618578/job/88905568590?pr=18115
```

--- a/.github/workflows/linux-build-base.yml
+++ b/.github/workflows/linux-build-base.yml
@@ -105,7 +105,7 @@ jobs:
     # check prevents errors when forks fast-forward their main branch.
     if: ${{ github.repository == 'facebookincubator/velox' }}
     runs-on: 32-core-ubuntu
-    container: ghcr.io/czentgr/czentgr-test:thrift-deps
+    container: ghcr.io/facebookincubator/velox-dev:adapters
     defaults:
       run:
         shell: bash
@@ -892,7 +892,7 @@ jobs:
     if: ${{ inputs.is-full-mode && github.repository == 'facebookincubator/velox' }}
     runs-on: 32-core-ubuntu
     container:
-      image: ghcr.io/czentgr/czentgr-test:thrift-deps
+      image: ghcr.io/facebookincubator/velox-dev:adapters
       volumes:
         - /usr:/host_usr
         - /opt:/host_opt

